ProxyAppsSuite/tests/Laghos.yaml

```diff
@@ -0,0 +1,250 @@
+    subtitle: "{{dim_runs.id}}_{{dim_runs.problem}}"
+    variables:
+         dim_runs:
+           - { id: test1, problem: '0', dim_flag: "-dim", dim_val: '2', rp: '3', ok: '', ot: '', tf: '0.75', s: '', assembly: "-fa"}
+           - { id: test2, problem: '0', dim_flag: "-dim", dim_val: '3', rp: '1', ok: '', ot: '', tf: '0.75', s: '', assembly: "-fa"}
+           - { id: test3, problem: '1', dim_flag: "-dim", dim_val: '2', rp: '3', ok: '', ot: '', tf: '0.8', s: '', assembly: "-fa"}
+           - { id: test4, problem: '1', dim_flag: "-dim", dim_val: '3', rp: '2', ok: '', ot: '', tf: '0.6', s: '', assembly: "-fa"}
+           - { id: test5, problem: '2', dim_flag: "-dim", dim_val: '1', rp: '5', ok: '', ot: '', tf: '0.2', s: '', assembly: "-fa"}
+           - { id: test6, problem: '3', dim_flag: "-m", dim_val: 'data/rectangle01_quad.mesh', rp: '2', ok: '', ot: '', tf: '3.0', s: '', assembly: "-fa"}
+           - { id: test7, problem: '3', dim_flag: "-m", dim_val: 'data/box01_hex.mesh', rp: '1', ok: '', ot: '', tf: '3.0', s: '', assembly: "-fa"}
+           - { id: test8, problem: '4', dim_flag: "-m", dim_val: "data/square_gresho.mesh", rp: '3', ok: '3', ot: '2', tf: '0.62831853', s: '7', assembly: "-pa"}
+    slurm:
+        num_nodes: 2
+    permute_on: dim_runs
+    run:
+        timeout: 2000
+        modules: [ gcc, openmpi/2.1.2 ]
+        cmds:
+            - cd Laghos
+            - if [[ {{dim_runs.problem}} -eq 4 ]]; then
+            -    srun -n 8 ./laghos --problem "{{dim_runs.problem}}" "{{dim_runs.dim_flag}}" "{{dim_runs.dim_val}}" -rp "{{dim_runs.rp}}" -ok "{{dim_runs.ok}}" -ot "{{dim_runs.ot}}" -tf "{{dim_runs.tf}}" -s "{{dim_runs.s}}" "{{dim_runs.assembly}}" &> {{dim_runs.id}}.out 
+            - else 
+            -    srun -n 8 ./laghos --problem "{{dim_runs.problem}}" "{{dim_runs.dim_flag}}" "{{dim_runs.dim_val}}" -rp "{{dim_runs.rp}}" -tf "{{dim_runs.tf}}" "{{dim_runs.assembly}}" &> {{dim_runs.id}}.out
+            - fi
+            - cd ..
+            - mv Laghos/{{dim_runs.id}}.out {{dim_runs.id}}.out
+            - cat {{dim_runs.id}}.out
+    build:
+        modules: [gcc, openmpi/2.1.2]
+        source_path: laghos
+        env:
+            CC: mpicc
+        cmds:
+            #HYPRE / METIS
+            - echo "beginning hypre and metis build"
+            - cd Laghos/
+            - make setup
+            - cd ../
+            #MFEM
+            - echo "beginning mfem build"
+            - cd mfem
+            - make parallel -j
+            - cd ../
+            #Laghos
+            - echo "beginning laghos build"
+            - cd Laghos
+            - make -j
+    result_parse:
+        constant:
+            test_id:
+                const: '{{dim_runs.id}}'
+        regex:
+            results:
+               regex: 'Major kernels total time.*'
+               action: 'store_true'
+               files: '{{dim_runs.id}}.out'
```

Comment:
You've made me realize I forgot to document a few things in 2.2, both of which will allow you to simply this whole section.

You have a couple options here. 
Option 1:
```yaml
result_parse:
  regex:
    result:
      regex: 'Major kernels total time.*'
```
Will work without the _action_ or _files_ arguments. 
 - You cat that file in the run script, which puts it in to the run.log, which is what the default for _files_ is anyway.
 - When storing to 'result' (which I assume you meant, instead of 'results'), 'store_true' is the default action.
   - That's 'forgot to document' thing 1.

Option 2:
```yaml
```yaml
result_parse:
  regex:
    _default:
      files: '{{dim_runs.id}}.out'
    result:
      regex: 'Major kernels total time.*'
      # Still dropping the action because its the default.
```

You can set defaults for every key under a result parser using the '_default' key. Sorry, forgot to document that too... :(